src/ngraph/opsets/opset.cpp

```diff
@@ -112,3 +112,24 @@ const ngraph::OpSet& ngraph::get_opset3()
     }
     return opset;
 }
+
+const ngraph::OpSet& ngraph::get_ie_opset()
+{
+    static std::mutex init_mutex;
+    static bool opset_is_initialized = false;
+    static OpSet opset;
+    if (!opset_is_initialized)
+    {
+        std::lock_guard<std::mutex> guard(init_mutex);
+        if (!opset_is_initialized)
+        {
+#define NGRAPH_OP(NAME, NAMESPACE) opset.insert<NAMESPACE::NAME>();
+#include "ngraph/opsets/opset1_tbl.hpp"
+#include "ngraph/opsets/opset2_tbl.hpp"
+#include "ngraph/opsets/opset3_tbl.hpp"
```

Comment:
I'm not sure whether this is correct. IE engine opset is always the highest. Opset `X` contains ops from opset `X-1` and **replaces** some ops with their new versions. With the above code the Opset's member `m_op_types` would have more elements than `m_name_type_info_map`. Since the higher opset versions would overwrite operators with same names.  This way one can call ie.:  `contains_type(const NodeTypeInfo& type_info)` for operator from opset1 and would get `true`, while only `opset3` could be created (for example `Topk`).

Reply:
I guess it's not really an opset. What we really want is the union of the set of ops. Adding an accessor for `m_op_types` will let us form that. I'll make those changes.